packages/bsky/src/api/app/bsky/feed/getAuthorFeed.ts

```diff
@@ -1,112 +1,171 @@
-            'BlockedByActor',
-          )
-        }
-      }
-
       const actorService = ctx.services.actor(db)
       const feedService = ctx.services.feed(db)
       const graphService = ctx.services.graph(db)
+      const viewer =
+        auth.credentials.type === 'access' ? auth.credentials.did : null
 
-      // maybe resolve did first
-      const actorRes = await actorService.getActor(actor)
-      if (!actorRes) {
-        throw new InvalidRequestError('Profile not found')
-      }
-      const actorDid = actorRes.did
-
-      // defaults to posts, reposts, and replies
-      let feedItemsQb = feedService
-        .selectFeedItemQb()
-        .where('originatorDid', '=', actorDid)
-
-      if (filter === 'posts_with_media') {
-        feedItemsQb = feedItemsQb
-          // and only your own posts/reposts
-          .where('post.creator', '=', actorDid)
-          // only posts with media
-          .whereExists((qb) =>
-            qb
-              .selectFrom('post_embed_image')
-              .select('post_embed_image.postUri')
-              .whereRef('post_embed_image.postUri', '=', 'feed_item.postUri'),
-          )
-      } else if (filter === 'posts_no_replies') {
-        feedItemsQb = feedItemsQb.where((qb) =>
-          qb
-            .where('post.replyParent', 'is', null)
-            .orWhere('type', '=', 'repost'),
-        )
-      }
-
-      const keyset = new FeedKeyset(
-        ref('feed_item.sortAt'),
-        ref('feed_item.cid'),
-      )
-
-      feedItemsQb = paginate(feedItemsQb, {
-        limit,
-        cursor,
-        keyset,
-      })
-
-      const [feedItems, repoRev] = await Promise.all([
-        feedItemsQb.execute(),
+      const [result, repoRev] = await Promise.all([
+        getAuthorFeed(
+          { ...params, viewer },
+          { db, actorService, feedService, graphService },
+        ),
         actorService.getRepoRev(viewer),
       ])
-      setRepoRev(res, repoRev)
 
-      const feedItemsSafe = await graphService.filterBlocksAndMutes(feedItems, {
-        getBlockPairs(item) {
-          if (viewer) {
-            return [[viewer, item.postAuthorDid]]
-          }
-        },
-        getMutePairs(item) {
-          // Hide reposts of muted content
-          if (viewer && item.type === 'repost') {
-            return [[viewer, item.postAuthorDid]]
-          }
-        },
-      })
-
-      const feed = await feedService.hydrateFeed(feedItemsSafe, viewer)
+      setRepoRev(res, repoRev)
 
       return {
         encoding: 'application/json',
-        body: {
-          feed,
-          cursor: keyset.packFromResult(feedItems),
-        },
+        body: result,
       }
     },
   })
 }
+
+export const skeleton = async (
+  params: Params,
+  ctx: Context,
+): Promise<SkeletonState> => {
+  const { cursor, limit, actor, filter, viewer } = params
+  const { db, actorService, feedService, graphService } = ctx
+  const { ref } = db.db.dynamic
+
+  // maybe resolve did first
```

Comment:
if we wanted to, we could simplify this logic by just not returning block violating posts - ie an empty feed 🤷‍♂️ 

Tradeoff is:
- trimming off a couple of DB hits that we do on every req
- some extra composition/hydration for reqs on a blocked actor

Reply:
I'm totally down, would just need to be on the lookout for whether the client uses this error as a signal in any way.

Reply:
Ended-up punting on this, since the client does make use of it: https://github.com/bluesky-social/social-app/blob/565c1a0510b5fb516cc23db4f4e2ec903ad66730/src/state/models/feeds/posts.ts#L330-L331